src/lsptoolshost/roslynLanguageServer.ts

```diff
@@ -538,7 +558,40 @@ export class RoslynLanguageServer {
             childProcess = cp.spawn(serverPath, args, cpOptions);
         }
 
-        return childProcess;
+        // The server process will create the named pipe used for communcation. Wait for it to be created,
+        // and listen for the server to pass back the connection information via stdout.
+        const namedPipeConnectionPromise = new Promise<TransmittedInformation>((resolve) => {
+            childProcess.stdout.on('data', (data: { toString: (arg0: any) => any }) => {
+                const result: string = isString(data) ? data : data.toString(RoslynLanguageServer.encoding);
+                _channel.append('[stdout] ' + result);
+
+                // Use the regular expression to find all JSON lines
+                const jsonLines = result.match(RoslynLanguageServer.namedPipeKeyRegex);
+                if (jsonLines) {
+                    const transmittedPipeNameInfo: TransmittedInformation = JSON.parse(jsonLines[0]);
+                    _channel.appendLine('received named pipe information from server');
+                    resolve(transmittedPipeNameInfo);
+                }
+            });
+        });
+
+        // Wait for the server to send back the name of the pipe to connect to.
+        _channel.appendLine('waiting for named pipe information from server...');
+        const pipeConnectionInfo = await namedPipeConnectionPromise;
```

Comment:
we should probably add a timeout here and show an error toast if it times out.

Reply:
Yes that's a good idea. Should probably do it for both receiving the name and making the named pipe connection

Reply:
Updated to throw an error, which will log. The "couldn't create connection" toast will show. 
![image](https://github.com/dotnet/vscode-csharp/assets/9122518/b7a99a1b-caef-46cf-b230-402a78a5b6be)